### PR TITLE
Fix cumprod to work properly with Integer columns.

### PR DESCRIPTION
**Pull request by itholic**
Basically, this PR addressed https://github.com/databricks/koalas/pull/1739#pullrequestreview-482833894

`cumprod` for `DataFrame` & `Series` & `GroupBy` isn't working properly with integer columns.

```python
>>> pdf
     A    B  C
0  2.0  1.0  1
1  5.0  NaN  2
2  1.0  1.0  3
3  2.0  4.0  4
4  4.0  9.0  5

>>> pdf.cumprod()
      A     B    C
0   2.0   1.0    1
1  10.0   NaN    2
2  10.0   1.0    6
3  20.0   4.0   24
4  80.0  36.0  120

>>> ks.from_pandas(pdf).cumprod()
      A     B      C
0   2.0   1.0    1.0
1  10.0   NaN    2.0
2  10.0   1.0    6.0
3  20.0   4.0   24.0
4  80.0  36.0  120.0
```

This PR addressed it and also addressed the related tests.

```python
>>> pdf.cumprod()
             A     B    C
0.986323   2.0   1.0    1
0.297507  10.0   NaN    2
0.617855  10.0   1.0    6
0.711719  20.0   4.0   24
0.290114  80.0  36.0  120

>>> ks.from_pandas(pdf).cumprod()
             A     B    C
0.986323   2.0   1.0    1
0.297507  10.0   NaN    2
0.617855  10.0   1.0    6
0.711719  20.0   4.0   24
0.290114  80.0  36.0  120
```

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -5139,7 +5141,10 @@ def negative_check(s):
             return F.sum(F.log(negative_check(scol)))
 
         kser = self._cum(cumprod, skipna, part_cols)
-        return kser._with_new_scol(F.exp(kser.spark.column))
+        result = kser._with_new_scol(F.exp(kser.spark.column))
+        if isinstance(data_type, LongType):
```
Comment (ueshin) [inline databricks/koalas/series.py:5145]: Shall we make the type checking the same as the other similar functions?

Comment (ueshin) [reply databricks/koalas/series.py:5145]: Indeed, we should handle `IntegerType`, `ShortType`, and `ByteType` as well.

Comment (itholic) [reply databricks/koalas/series.py]: Thanks, @ueshin !!

File: databricks/koalas/tests/test_dataframe.py [hunk 2]
```diff
@@ -2264,8 +2264,8 @@ def _test_cumprod(self, pdf, kdf):
 
     def test_cumprod(self):
         pdf = pd.DataFrame(
-            [[2.0, 1.0], [5, None], [1.0, 1.0], [2.0, 4.0], [4.0, 9.0]],
-            columns=list("AB"),
+            [[2.0, 1.0, 1], [5, None, 2], [1.0, 1.0, 3], [2.0, 4.0, 4], [4.0, 9.0, 5]],
+            columns=list("ABC"),
             index=np.random.rand(5),
         )
```
Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py:2270]: Shall we use different inputs for `PySpark < 2.4` where `transpose` won't work with different data types.

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: databricks/koalas/tests/test_dataframe.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/tests/test_dataframe.py]: Thanks, @ueshin !

File: databricks/koalas/tests/test_dataframe.py [hunk 3]
```diff
@@ -2263,13 +2263,23 @@ def _test_cumprod(self, pdf, kdf):
...
+        if LooseVersion(pyspark.__version__) >= LooseVersion("2.4"):
+            pdf = pd.DataFrame(
+                [[2.0, 1.0, 1], [5, None, 2], [1.0, 1.0, 3], [2.0, 4.0, 4], [4.0, 9.0, 5]],
+                columns=list("ABC"),
+                index=np.random.rand(5),
+            )
+            kdf = ks.from_pandas(pdf)
+            self._test_cumprod(pdf, kdf)
+        else:
+            with self.sql_conf({"spark.sql.execution.arrow.enabled": False}):
```
Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py:2275]: Why is this needed?

Comment (itholic) [reply databricks/koalas/tests/test_dataframe.py:2275]: Think it was my mistaken. Thanks for the point out

**Review by HyukjinKwon — APPROVED** [review]
